### ISSUE TITLE
"isBlackBox()" incorrect 

### ISSUE THREAD

**Issue by Licheng-Guo** [Xilinx/RapidWright#155]
Hi Chris,

I'm trying the `populateBlackBox()` method in `DesignTools`. The designs are generated in Vivado.

RW complains that the target cell is not a black box, but it is indeed a black box seen from Vivado. 

I'm trying this because `read_checkpoint -cell` in vivado will take forever...

Could you please help take a look? I attach a 4-line code and the checkpoints to repeat the error.
`ERROR: The cell instance CR_X0Y0_To_CR_X1Y1_routing_U0/CR_X0Y0_To_CR_X1Y1_U0 is not a black box.`

```java
package com.xilinx.rapidwright.examples;

import com.xilinx.rapidwright.design.Design;
import com.xilinx.rapidwright.design.DesignTools;

public class mergeBlackBoxDCPs {
  public static void main(String[] args) {
    Design top = Design.readCheckpoint("./kernel3_kernel3.dcp");    
    Design slot = Design.readCheckpoint("./CR_X0Y0_To_CR_X1Y1_routed_free_run_compute.dcp");
    DesignTools.populateBlackBox(top, "CR_X0Y0_To_CR_X1Y1_routing_U0/CR_X0Y0_To_CR_X1Y1_U0", slot);
    top.writeCheckpoint("test.dcp");
  }
}
```

The checkpoints could be found at:
https://drive.google.com/drive/folders/1GSgURFyteVfEsKF1W9qzDlC2EQEJfTfU?usp=sharing

Great thanks for your help!

-Licheng

**Comment by clavin-xlnx** [comment]
Hi Licheng,

I was able to run with your design, I didn't reproduce the same error you received, but I resolved two other issues that I ran into.  First, I got an NPE because the GND net was missing from one of your designs, I've updated the code to include a fix to correctly handle that scenario.

Second, there was an issue with populating the netlist and moving the EDIFCells across libraries correctly, this has also been resolved.  To proceed, I suggest the following:

1. I'm not sure how you got the error you did, with the designs you sent me, that did not happen.  So please pull the latest changes and make sure you are on the latest version of RapidWright and you are reading the designs you intend.
2. This method is not fully tested (as noted in the Javadoc).  So, it may not work under all conditions, but please report back any issues you find.
3. It appears your design had some encrypted IP (floating point operators) that were not included with your example.  That is ok, but I wasn't able to fully re-assemble the design.  RapidWright can handle this, but when you write out the DCP, it should create a Tcl script that will be needed in order to load the DCP back into Vivado.  Be sure to run the script instead of the usual `open_checkpoint` command.

Thanks, and let us know if you have more trouble.

Chris

**Comment by Licheng-Guo** [comment]
Hi Chris,

Thanks a lot for your effort!

I just found another case that will cause an NPE:

```java
Design top = Design.readCheckpoint("./kernel3_kernel3.dcp");    

Design CR_X6Y6_To_CR_X7Y7 = Design.readCheckpoint("./CR_X6Y6_To_CR_X7Y7_routed_free_run_compute/CR_X6Y6_To_CR_X7Y7_routed_free_run_compute.dcp");
DesignTools.populateBlackBox(top, "CR_X6Y6_To_CR_X7Y7_routing_U0/CR_X6Y6_To_CR_X7Y7_U0", CR_X6Y6_To_CR_X7Y7);
Design CR_X4Y6_To_CR_X5Y7 = Design.readCheckpoint("./CR_X4Y6_To_CR_X5Y7_routed_free_run_compute/CR_X4Y6_To_CR_X5Y7_routed_free_run_compute.dcp");
DesignTools.populateBlackBox(top, "CR_X4Y6_To_CR_X5Y7_routing_U0/CR_X4Y6_To_CR_X5Y7_U0", CR_X4Y6_To_CR_X5Y7);
```

```
Exception in thread "main" java.lang.NullPointerException
        at com.xilinx.rapidwright.edif.EDIFNetlist.getNetAliases(EDIFNetlist.java:1042)
        at com.xilinx.rapidwright.edif.EDIFNetlist.generateParentNetMap(EDIFNetlist.java:1157)
        at com.xilinx.rapidwright.edif.EDIFNetlist.getParentNetMap(EDIFNetlist.java:1099)
        at com.xilinx.rapidwright.edif.EDIFNetlist.getParentNetName(EDIFNetlist.java:1094)
        at com.xilinx.rapidwright.design.DesignTools.postBlackBoxCleanup(DesignTools.java:882)
        at com.xilinx.rapidwright.design.DesignTools.populateBlackBox(DesignTools.java:860)
        at com.xilinx.rapidwright.ap.mergeBlackBoxDCPs.main(mergeBlackBoxDCPs.java:13)
```

I upload the new files to the same drive:
https://drive.google.com/drive/folders/1GSgURFyteVfEsKF1W9qzDlC2EQEJfTfU?usp=sharing

I'm not sure if anything is wrong with my checkpoints? Those are generated by Vivado in out-of-context mode. Though I could correctly assemble them in Vivado.

Thanks again!

Best,
Licheng

**Comment by clavin-xlnx** [comment]
Hi Licheng,

I was unable to reproduce your error.  Perhaps you might have some old/stale files in your environment?  Try performing a "clean" and check your CLASSPATH variable to make sure you don't have any old files being referenced.  You may want to even try a fresh install of RapidWright that is separate just to be sure.   I am using the latest version of the code in GitHub and I have the latest jars (from the 2020.2.3 release).  Let me know if you still have an issue.

Chris

**Comment by clavin-xlnx** [comment]
In some of our more recent efforts, I did run into this issue and I believe I have resolved with some recent changes (should be included in 2020.2.6).  Please reopen if we run into it again.
